nifi-extension-bundles/nifi-box-bundle/nifi-box-processors/src/test/java/org/apache/nifi/processors/box/CreateBoxMetadataTemplateTest.java

```diff
@@ -0,0 +1,219 @@
+    private void configureJsonRecordReader(TestRunner runner) throws InitializationException {
+        final JsonTreeReader readerService = new JsonTreeReader();
+        runner.addControllerService("json-reader", readerService);
+        runner.enableControllerService(readerService);
+    }
+
+    @Test
+    void testSuccessfulTemplateCreation() {
+        final String inputJson = """
+                [
+                    {"key": "field1", "type": "string", "displayName": "Field One"},
+                    {"key": "field2", "type": "float"}
+                ]
+                """;
+
+        testRunner.enqueue(inputJson);
+        testRunner.run();
+        assertEquals(2, capturedFields.size());
+
+        MetadataTemplate.Field field1 = capturedFields.getFirst();
+        assertEquals("field1", field1.getKey());
+        assertEquals("string", field1.getType());
+        assertEquals("Field One", field1.getDisplayName());
+
+        MetadataTemplate.Field field2 = capturedFields.get(1);
+        assertEquals("field2", field2.getKey());
+        assertEquals("float", field2.getType());
+
+        testRunner.assertAllFlowFilesTransferred(CreateBoxMetadataTemplate.REL_SUCCESS, 1);
+        final MockFlowFile flowFile = testRunner.getFlowFilesForRelationship(CreateBoxMetadataTemplate.REL_SUCCESS).get(0);
+        flowFile.assertAttributeEquals("box.template.name", TEMPLATE_NAME);
+        flowFile.assertAttributeEquals("box.template.key", TEMPLATE_KEY);
+        flowFile.assertAttributeEquals("box.template.scope", CreateBoxMetadataTemplate.SCOPE_ENTERPRISE);
+        flowFile.assertAttributeEquals("box.template.fields.count", "2");
+    }
+
+    @Test
+    void testEmptyInput() {
+        final String inputJson = "[]";
+
+        testRunner.enqueue(inputJson);
+        testRunner.run();
+
+        testRunner.assertAllFlowFilesTransferred(CreateBoxMetadataTemplate.REL_FAILURE, 1);
+        final MockFlowFile flowFile = testRunner.getFlowFilesForRelationship(CreateBoxMetadataTemplate.REL_FAILURE).getFirst();
+        flowFile.assertAttributeEquals("error.message", "No valid metadata field specifications found in the input");
+    }
+
+    @Test
+    void testInvalidRecords() {
+        // First record missing the key; second record has an invalid type.
+        final String inputJson = """
+                [
+                    {"type": "string", "displayName": "No Key"},
+                    {"key": "field2", "type": "invalid"}
+                ]
+                """;
+
+        testRunner.enqueue(inputJson);
+        testRunner.run();
+
+        testRunner.assertAllFlowFilesTransferred(CreateBoxMetadataTemplate.REL_FAILURE, 1);
+        final MockFlowFile flowFile = testRunner.getFlowFilesForRelationship(CreateBoxMetadataTemplate.REL_FAILURE).getFirst();
+        final String errorMessage = flowFile.getAttribute("error.message");
+        assertTrue(errorMessage.contains("missing a key field") || errorMessage.contains("has an invalid type"));
```

Comment:
I assume we want to ensure that both error messages are present here, right?
Then it has to be 2 separate assertions.
```suggestion
assertTrue(errorMessage.contains("has an invalid type"));
assertTrue(errorMessage.contains("missing a key field"));
```